AxonDeepSeg/segment.py

```diff
@@ -375,176 +240,45 @@ def main(argv=None):
-            + 'but also increase the segmentation time. \n'
-            + 'Default value: '+str(default_overlap)+'\n'
-            + 'Recommended range of values: [10-100]. \n',
-        default=None
-    )
-    ap.add_argument(
-        "-z", "--zoom", 
-        required=False, 
-        help='Zoom factor. \n'
-            + 'When applying the model, the pixel size of the image will be \n'
-            + 'multiplied by this number.',
-        default=None,
-    )
-    ap.add_argument(
-        "-r", "--sweeprange",
-        nargs=2,
-        metavar=('LOWER', 'UPPER'),
-        required=False,
-        help='Lower and upper bounds of zoom factor values to sweep.',
-        default=None,
-    )
-    ap.add_argument(
-        "-l", "--sweeplength",
-        required=False,
-        type=int,
-        help='Number of zoom factor values to be computed by the sweep.',
-        default=None,
-    )
-    ap.add_argument(
-        "--no-patch",
-        dest="no_patch",
-        action='store_true',
-        required=False,
-        help='Flag to segment the image without using patches. \n'
-             'The "--no-patch" flag supersedes the "--overlap" flag. \n'
-             'This option may not be suitable with large images depending on computer RAM capacity.'
-    )
     ap.add_argument(
         "--gpu-id",
         dest="gpu_id",
         required=False,
         type=int,
-        help='Number representing the GPU ID for segmentation if available. Default: 0.',
-        default=0,
+        help='Number representing the GPU ID for segmentation if available. Default: -1 (cpu).',
+        default=-1,
     )
     ap._action_groups.reverse()
 
     # Processing the arguments
     args = vars(ap.parse_args(argv))
-    type_ = str(args["type"])
     verbosity_level = int(args["verbose"])
-    overlap_value = [int(args["overlap"]), int(args["overlap"])] if args["overlap"] else None
-    if args["sizepixel"] is not None:
-        psm = float(args["sizepixel"])
-    else:
-        psm = None
     path_target_list = [Path(p) for p in args["imgpath"]]
-    new_path = Path(args["model"]) if args["model"] else None 
-    if args["zoom"] is not None:
-        zoom_factor = float(args["zoom"])
-    else:
-        zoom_factor = 1.0
-    no_patch = bool(args["no_patch"])
+    path_model = Path(args["model"]) if args["model"] else DEFAULT_MODEL_PATH
     gpu_id = int(args["gpu_id"])
 
     # Check for available GPU IDs
-    ads.check_available_gpus(gpu_id)
-
-    # Check for sweep mode
-    sweep_mode = (args["sweeprange"] is not None) & (args["sweeplength"] is not None)
-    if sweep_mode:
-        sweep_range = [float(args["sweeprange"][0]), float(args["sweeprange"][1])]
-        sweep_length = int(args["sweeplength"])
-        if len(path_target_list) != 1:
-            logger.error("Error: Please use a single image for zoom factor sweep.")
-            sys.exit(5)
+    if gpu_id >= 0:
+        ads.check_available_gpus(gpu_id)
```

Comment:
self: line not covered by tests

Reply:
Covered by b510948

Reply:
This breaks inference for me. Now, when `--gpu-id` is `None`, you put it manually to 0. However, in `axon_segmentation()`, we have this logic:

```python
predictor = nnUNetPredictor(
        perform_everything_on_gpu=True if gpu_id >= 0 else False,
        device=torch.device('cuda', gpu_id) if gpu_id >= 0 else torch.device('cpu'),
)
```

As such, because you are feeding `gpu-id=0` instead of `-1`, it tries to use CUDA but since I don't have an NVIDIA card in my laptop, it crashes:
```
RuntimeError: CUDA error: CUDA driver version is insufficient for CUDA runtime version
CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
Device-side assertions were explicitly omitted for this error check; the error probably arose while initializing the DSA handlers.
```

I'm actually not even sure how tests are passing.

Reply:
@hermancollin on my laptop (which doesn't have a CUDA GPU), it appears that `device=torch.device('cuda', gpu_id)` doesn't error when given `gpu_id=0`. So maybe it always defaults to CPU if you try to run it on a GPU but you don't have one?

<img width="685" alt="Screenshot 2024-08-16 at 12 37 49 PM" src="https://github.com/user-attachments/assets/444e89fb-dee6-4641-b786-8d408b929500">

If this is the case, then I think the only bug in my implementation would be if you were on a GPU but wanted to run on the CPU?

Reply:
@hermancollin Did you miss my reply here? Maybe this is why you were getting an error, were you testing on a GPU device? I was only running my tests locally (macbookpro) and on the GitHub action instances, neither of which have GPU.

Nonetheless, the easiest solution might be just to rewind back to your initial implementation, I'll do that now!